connection.go

```diff
@@ -93,3 +94,134 @@ func (e *EurekaConnection) UpdateApp(app *Application) {
 		}
 	}()
 }
+
+// AppUpdate is the outcome of an attempt to get a fresh snapshot of a Eureka
+// application's state, together with an error that may have occurred in that
+// attempt. If the Err field is nil, the App field will be non-nil.
+type AppUpdate struct {
+	App *Application
+	Err error
+}
+
+func sendAppUpdatesEvery(d time.Duration, produce func() AppUpdate, c chan<- AppUpdate, done <-chan struct{}) {
+	t := time.NewTicker(d)
+	defer t.Stop()
+	for {
+		select {
+		case <-done:
+			close(c)
+			return
+		case <-t.C:
+			// Drop attempted sends when the consumer hasn't received the last buffered update.
+			select {
+			case c <- produce():
+			default:
+			}
+		}
+	}
+}
+
+// ScheduleAppUpdates starts polling for updates to the Eureka application with
+// the given name, using the connection's configured polling interval as its
+// period. It sends the outcome of each update attempt to the returned channel,
+// and continues until the supplied done channel is either closed or has a value
+// available. Once done sending updates to the returned channel, it closes it.
+//
+// If prime is true, it sends at least one application update outcome to the
+// returned channel before returning.
+func (e *EurekaConnection) ScheduleAppUpdates(name string, prime bool, done <-chan struct{}) <-chan AppUpdate {
+	produce := func() AppUpdate {
+		app, err := e.GetApp(name)
+		return AppUpdate{app, err}
+	}
+	c := make(chan AppUpdate, 1)
+	if prime {
+		c <- produce()
+	}
+	go sendAppUpdatesEvery(time.Duration(e.PollInterval)*time.Second, produce, c, done)
+	return c
+}
+
+// An AppSource holds a periodically updated copy of a Eureka application.
+type AppSource struct {
+	m    *sync.RWMutex
+	app  *Application
+	done chan<- struct{}
+}
+
+// NewAppSource returns a new AppSource that offers a periodically updated copy
+// of the Eureka application with the given name, using the connection's
+// configured polling interval as its period.
+//
+// If prime is true, it waits for the first application update to complete
+// before returning, though it's possible that that first update attempt could
+// fail, so that a subsequent call to CopyLatestAppTo would return false.
+func (e *EurekaConnection) NewAppSource(name string, prime bool) *AppSource {
+	done := make(chan struct{})
+	updates := e.ScheduleAppUpdates(name, prime, done)
+	s := &AppSource{
+		done: done,
+	}
+	if prime {
+		if u := <-updates; u.Err != nil {
+			s.app = u.App
+		}
+	}
+	go func() {
+		for u := range updates {
+			if u.Err != nil {
```

Comment:
There's an inconsistency with the documentation for `Latest` at line 183, which says that if the most recent update attempt fails, we do want to assign `nil` to `u.App` here—and later return `nil` from `Latest`. Here, though, since we ignore failed updates—where an error arose—we won't wind up assigning `nil` to `s.app` unless `GetApp` can return a nil `Application` pointer with no corresponding `error`. My reading of `GetApp` says that it _won't_ return a nil `Application` pointer when the corresponding `error` is nil, meaning that we'd only ever assign a non-nil `Application` pointer to `s.app` here.

Given that, I can see two choices:
-  Remove this guard condition here, and _always_ take the latest `Application` pointer, even if it's nil, making it consistent with the documentation for `Latest`.
- Guard on `u.App` not being nil here, as you suggest, and amend the documentation for `Latest` to indicate that once `Latest` returns non-nil, it will never return nil again. Unfortunately, it can still return nil if no update attempts have succeeded yet.

I can see callers wanting a kind of self-preservation here, where if we had a good `Application` in hand at time T1, and later at time T2 we fail to update it, and at T3 we ask for the latest `Application`, we might like to have the "last known good" one still available to us. We could make this an option on `NewAppSource`—a boolean `preserveLastGood` parameter.

I'd appreciate your feedback.